src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/WorkOnReleaseEntityMapper.kt

```diff
@@ -1,35 +1,119 @@
-  @Mapping(target = "lastUpdatedAtPrison", source = "updatedAtPrison")
-  fun fromEntityToDomain(persistedEntity: WorkOnReleaseEntity?): WorkOnRelease
-
-  @ExcludeJpaManagedFieldsIncludingDisplayNameFields
-  @ExcludeReferenceField
-  @Mapping(target = "createdAtPrison", ignore = true)
-  @Mapping(target = "updatedAtPrison", source = "prisonId")
-  fun updateExistingEntityFromDto(@MappingTarget entity: WorkOnReleaseEntity, dto: UpdateWorkOnReleaseDto?)
+import java.util.UUID
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.AffectAbilityToWork as AffectAbilityToWorkEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.HopingToWork as HopingToWorkEntity
+
+@Component
+class WorkOnReleaseEntityMapper {
+
+  fun fromCreateDtoToEntity(dto: CreateWorkOnReleaseDto): WorkOnReleaseEntity {
+    val workOnReleaseEntity = WorkOnReleaseEntity()
+    workOnReleaseEntity.createdAtPrison = dto.prisonId
+    workOnReleaseEntity.updatedAtPrison = dto.prisonId
+    workOnReleaseEntity.hopingToWork = toHopingToWork(dto.hopingToWork)
+    workOnReleaseEntity.affectAbilityToWork = toEntity(dto.affectAbilityToWork)
+    workOnReleaseEntity.affectAbilityToWorkOther = dto.affectAbilityToWorkOther
+
+    workOnReleaseEntity.reference = UUID.randomUUID()
+
+    return workOnReleaseEntity
+  }
+
+  fun fromEntityToDomain(entity: WorkOnReleaseEntity): WorkOnRelease {
+    with(entity) {
+      return WorkOnRelease(
+        reference!!,
+        toHopingToWork(hopingToWork!!),
+        toModel(affectAbilityToWork),
+        affectAbilityToWorkOther,
+        createdBy,
+        createdAt,
+        createdAtPrison!!,
+        updatedBy,
+        updatedAt,
+        updatedAtPrison!!,
+      )
+    }
+  }
+
+  fun updateExistingEntityFromDto(entity: WorkOnReleaseEntity, dto: UpdateWorkOnReleaseDto?) {
+    if (dto == null) {
+      return
+    }
+    entity.updatedAtPrison = dto.prisonId
+    entity.hopingToWork = toHopingToWork(dto.hopingToWork)
+
+    val updatedAffectAbilityToWorkList = toEntity(dto.affectAbilityToWork)
+    if (entity.affectAbilityToWork != null) {
+      entity.affectAbilityToWork!!.clear()
+      entity.affectAbilityToWork!!.addAll(updatedAffectAbilityToWorkList)
+    } else {
+      entity.affectAbilityToWork = updatedAffectAbilityToWorkList
+    }
+    entity.affectAbilityToWorkOther = dto.affectAbilityToWorkOther
+  }
+
+  fun toEntity(list: List<AffectAbilityToWork>): MutableList<AffectAbilityToWorkEntity> {
+    return list.mapTo(ArrayList(list.size)) { affectAbilityToWorkToAffectAbilityToWork(it) }
+  }
+
+  fun toModel(list: List<AffectAbilityToWorkEntity>?): List<AffectAbilityToWork> {
+    return list?.map { toAffectAbilityToWork(it) } ?: emptyList()
+  }
+
+  fun toAffectAbilityToWork(affectAbilityToWork: uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.AffectAbilityToWork): AffectAbilityToWork {
```

Comment:
Minor, but wonder if it would be worth using an import alias here (eg: `AffectAbilityToWorkEntity`) rather than fully qualifying it?